### PR TITLE
Add a pet

### PR DESCRIPTION
**Pull request by benfox1216**
-Created pets in the database and models for them, and created a one-to-many relationship between pets and users (user has_many pets, pets belong_to user)
-Added the feature and model tests, including sad path testing of a pet not being created
-Added the routes and controller methods for the new page, and the pet creation, redirecting to the user profile
-Made sure the new pet was showing up on the user profile page

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -51,6 +52,8 @@ group :development, :test do
   gem 'pry'
   gem 'launchy'
   gem'shoulda-matchers'
+  gem 'factory_bot_rails'
+  gem 'faker'
 end
```
Comment (StephanieFriend) [inline Gemfile:57]: thank you for taking the time to create these!

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -7,7 +7,24 @@
     <%= text_field_tag(:zipcode) %>
     <%= submit_tag("Submit") %>
   <% end %>
-  <% else %>
-  <p>Zipcode: <%= @user.zipcode %></p>
+<% else %>
+  <p>Zipcode: <%= @user.zipcode %>
 <% end %>
```
Comment (StephanieFriend) [inline app/views/users/show.html.erb:12]: uhhhh, any reason why you got rid of a closing tag?

File: db/migrate/20200530223635_create_pets.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class CreatePets < ActiveRecord::Migration[5.1]
+  def change
+    create_table :pets do |t|
+      t.references :user, foreign_key: true
+      t.string :name
+      t.string :species
+      t.integer :age
+      t.string :breed
+      t.string :size
```
Comment (StephanieFriend) [inline db/migrate/20200530223635_create_pets.rb:9]: wern't we going to do an about me section for pets? And maybe make age optional?

File: app/views/pets/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  <%= f.text_field :name %>
+  <%= f.label :species %>
+  <%= f.text_area :species %>
+  <%= f.label :age %>
+  <%= f.text_field :age %>
+  <%= f.label :breed %>
+  <%= f.text_field :breed %>
+  <%= f.label :size %>
+  <%= f.text_field :size %>
+
```
Comment (StephanieFriend) [inline app/views/pets/new.html.erb:14]: maybe do a dropdown for size?

File: spec/features/user/user_can_add_a_pet_spec.rb [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+describe "As a user" do
+  it "I can add a pet" do
+    # user = create(:user)
+    
+    user = User.create(name: "Bob", email: "bob@example",
+                        image: "https://upload.wikimedia.org/wikipedia/commons/thumb/9/94/Stick_Figure.svg/1200px-Stick_Figure.svg.png",
+                        google_token: "abc123", zipcode: "80012")
+    #allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
```
Comment (StephanieFriend) [inline spec/features/user/user_can_add_a_pet_spec.rb:11]: lets clean this back up

File: spec/features/user/user_can_add_a_pet_spec.rb [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    end
+  end
+  
+  it "I will get a flash message if a pet is not created" do
+    user = create(:user)
+    
+    allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+    
+    visit "/profile/#{user.id}"
+    save_and_open_page
```
Comment (StephanieFriend) [inline spec/features/user/user_can_add_a_pet_spec.rb:68]: dont forget to remove the save and opens

File: app/views/users/show.html.erb (hunk 2 above)
Comment (benfox1216) [reply app/views/users/show.html.erb:12]: Definitely an accident, my bad.

File: db/migrate/20200530223635_create_pets.rb (hunk 3 above)
Comment (benfox1216) [reply db/migrate/20200530223635_create_pets.rb:9]: Yes that’s true. And I’m fine with age being optional. We should probably all discuss the attributes we’ll ask for, and which ones will be required, but for now I’ll just make it work.

File: app/views/pets/new.html.erb (hunk 4 above)
Comment (benfox1216) [reply app/views/pets/new.html.erb:14]: For sure, I can do that.

File: spec/features/user/user_can_add_a_pet_spec.rb (hunk 6 above)
Comment (benfox1216) [reply spec/features/user/user_can_add_a_pet_spec.rb:68]: Oh good catch!

File: spec/features/user/user_can_add_a_pet_spec.rb (hunk 5 above)
Comment (benfox1216) [reply spec/features/user/user_can_add_a_pet_spec.rb:11]: Looks like I fixed this, but after I pushed up to this branch again. Will be in next push.

File: spec/features/user/user_can_add_a_pet_spec.rb (hunk 6 above)
Comment (benfox1216) [reply spec/features/user/user_can_add_a_pet_spec.rb:68]: Same here, already removed but not included in the push.

File: app/views/pets/new.html.erb (hunk 4 above)
Comment (StephanieFriend) [reply app/views/pets/new.html.erb:16]: Looks awesome!

File: db/migrate/20200530223635_create_pets.rb (hunk 3 above)
Comment (StephanieFriend) [reply db/migrate/20200530223635_create_pets.rb:9]: Sounds good to me. :)

File: spec/features/user/user_can_add_a_pet_spec.rb (hunk 5 above)
Comment (StephanieFriend) [reply spec/features/user/user_can_add_a_pet_spec.rb:7]: Thank you so much!

**Pull request merged**